src/client/components/responsive-video/responsive-video.vue

```diff
@@ -1,17 +1,113 @@
 <template>
-  <figure>
-    <a :href="video.url" target="_blank" rel="noopener" @click="play()">
-      <img alt="" :src="video.thumbnailUrl" :width="video.width" :height="video.height" />
-    </a>
-    <figcaption v-if="video.title">{{ video.title }}</figcaption>
-  </figure>
+  <div class="responsive-video">
+    <figure>
+      <fixed-ratio class="responsive-video__canvas" :width="video.width" :height="video.height">
+        <lazy-load>
+          <img
+            class="responsive-video__image"
+            v-if="!isPlaying"
+            alt=""
+            :src="video.thumbnailUrl"
+          />
+        </lazy-load>
+        <iframe
+          v-if="isPlaying"
+          :src="videoSource"
+          style="width:100%; height:100%;"
+          frameborder="0"
+          webkitallowfullscreen
+          mozallowfullscreen
+          allowfullscreen
+          allow="autoplay">
+        </iframe>
+        <a :href="video.url" v-if="!isPlaying" class="responsive-video__button" @click="play()">
+          <span class="a11y-sr-only">Play</span>
+          <img class="responsive-video__icon" src="/images/play.svg" />
+        </a>
+      </fixed-ratio>
+      <figcaption v-if="video.title">
+        <a :href="video.url" target="_blank" rel="noopener">
+          {{ video.title }}
+        </a>
+      </figcaption>
+    </figure>
+  </div>
 </template>
 
 <script>
+
+import FixedRatio from '../fixed-ratio'
+import LazyLoad from '../lazy-load'
+
 export default {
   props: ['video'],
+  components: { FixedRatio, LazyLoad },
+  computed: {
+    videoSource() {
+      if (!this.isPlaying) return ''
+      switch (this.video.provider) {
+        case 'vimeo':
+          return this.isPlaying ? `https://player.vimeo.com/video/${this.video.providerUid}?autoplay=1` : ''
```

Comment:
You already `return`ed in case of `!this.isPlaying`. So here you can simply directly `return 'https://...'`.